### PR TITLE
Dpp 1119 consolidated production merge

### PR DESCRIPTION
**Pull request by carlbm**

### CODE REVIEW

File: src/OpaEarning/Build/Deploy/1718/copy mappings/DasEarningsCopyToDedsMapping.xml [hunk 1]
```diff
@@ -2,11 +2,6 @@
 <CopyCatConfig>
```
Comment (davehowlett) [inline src/OpaEarning/Build/Deploy/1718/copy mappings/DasEarningsCopyToDedsMapping.xml:2]: this file should be git ignored as it is build output

File: src/SFA.DAS.PaymentComponents.sln [hunk 2]
```diff
@@ -226,7 +226,7 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Migration Scripts", "Migrat
 		SharedPipelineComponents\DataLockEvents\DeployDataLock\sql\migrationscripts\002_DataLockEvents_Change_Commitment_Verion.sql = SharedPipelineComponents\DataLockEvents\DeployDataLock\sql\migrationscripts\002_DataLockEvents_Change_Commitment_Verion.sql
 		SharedPipelineComponents\DataLockEvents\DeployDataLock\sql\migrationscripts\003_DataLockEvents_Change_LearnRef_AimSequence_ColumnType.sql = SharedPipelineComponents\DataLockEvents\DeployDataLock\sql\migrationscripts\003_DataLockEvents_Change_LearnRef_AimSequence_ColumnType.sql
 		SharedPipelineComponents\DataLockEvents\DeployDataLock\sql\migrationscripts\004_DataLockEvents_Add_TransactionTypesFlag.sql = SharedPipelineComponents\DataLockEvents\DeployDataLock\sql\migrationscripts\004_DataLockEvents_Add_TransactionTypesFlag.sql
-		SharedPipelineComponents\DataLockEvents\DeployDataLock\sql\migrationscripts\005_DataLockEvents_Tables_Add_Indexes.sql = SharedPipelineComponents\DataLockEvents\DeployDataLock\sql\migrationscripts\005_DataLockEvents_Tables_Add_Indexes.sql
```
Comment (davehowlett) [inline src/SFA.DAS.PaymentComponents.sln:229]: file still appears in SFA.DAS.Provider.Events.DataLock.IntegrationTests/DbSetupScripts/MigrationScripts. Please remove if it shouldn't be there

File: src/SharedPipelineComponents/DataLockEvents/DeployDataLock/sql/ddl/datalockevents.transient.reference.ddl.tables.sql [hunk 3]
```diff
@@ -47,6 +47,9 @@ CREATE TABLE [Reference].[DataLockEvents]
 )
 GO
 
+CREATE INDEX IX_ReferenceDataLockEvents_DataLockEventId ON Reference.DataLockEvents (DataLockEventId)
+
```
Comment (davehowlett) [inline src/SharedPipelineComponents/DataLockEvents/DeployDataLock/sql/ddl/datalockevents.transient.reference.ddl.tables.sql:51]: Should there be a GO after this?

File: src/SFA.DAS.PaymentComponents.sln (hunk 2 above)
Comment (davehowlett) [reply src/SFA.DAS.PaymentComponents.sln:229]: should the 005_DataLockEvents_Add_Indexes.sql file be added in its place?

File: src/SharedPipelineComponents/Datalock/DeployPeriodEnd/sql/dml/05 PeriodEnd.DataLock.Populate.Reference.DataLockPriceEpisode.dml.sql [hunk 4]
```diff
@@ -17,6 +17,12 @@ BEGIN
 END
 GO
 
+
+IF EXISTS (SELECT NULL FROM sys.indexes WHERE name='IX_DataLockPriceEpisode_Uln')
+BEGIN
+	DROP INDEX IX_DataLockPriceEpisode_Uln ON Reference.DataLockPriceEpisode
+END
+
```
Comment (davehowlett) [inline src/SharedPipelineComponents/Datalock/DeployPeriodEnd/sql/dml/05 PeriodEnd.DataLock.Populate.Reference.DataLockPriceEpisode.dml.sql:25]: Add a GO at the end

File: src/SharedPipelineComponents/Datalock/DeployPeriodEnd/sql/dml/05 PeriodEnd.DataLock.Populate.Reference.DataLockPriceEpisode.dml.sql [hunk 5]
```diff
@@ -91,4 +97,6 @@ CREATE CLUSTERED INDEX [IDX_DataLockPriceEpisode_Ukprn] ON [Reference].[DataLock
 )WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, SORT_IN_TEMPDB = OFF, DROP_EXISTING = OFF, ONLINE = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
 GO
 
-CREATE INDEX IX_DataLockPriceEpisode_DataLockEvents ON Reference.DataLockPriceEpisode (UKPRN, LearnRefNumber, PriceEpisodeIdentifier)
\ No newline at end of file
+CREATE INDEX IX_DataLockPriceEpisode_DataLockEvents ON Reference.DataLockPriceEpisode (UKPRN, LearnRefNumber, PriceEpisodeIdentifier)
+
```
Comment (davehowlett) [inline src/SharedPipelineComponents/Datalock/DeployPeriodEnd/sql/dml/05 PeriodEnd.DataLock.Populate.Reference.DataLockPriceEpisode.dml.sql]: Add a GO between these lines

File: src/SharedPipelineComponents/Datalock/SFA.DAS.CollectionEarnings.DataLock/DataLockProcessor.cs [hunk 6]
```diff
@@ -37,6 +37,7 @@ protected DataLockProcessor()
         public virtual void Process()
```
Comment (davehowlett) [inline src/SharedPipelineComponents/Datalock/SFA.DAS.CollectionEarnings.DataLock/DataLockProcessor.cs:35]: two using statements at the top of the file are unused and should be removed:

using System.Collections.Generic;
using System.Linq;

File: src/SharedPipelineComponents/Datalock/SFA.DAS.CollectionEarnings.DataLock/DataLockTask.cs [hunk 7]
```diff
@@ -8,9 +9,11 @@ namespace SFA.DAS.CollectionEarnings.DataLock
 {
     public class DataLockTask : DcfsTask
     {
-        private IDependencyResolver _dependencyResolver;
+        private readonly IDependencyResolver _dependencyResolver;
         private const string DataLockSchema = "DataLock";
 
+        public static int CommandTimeout = 180;
```
Comment (davehowlett) [inline src/SharedPipelineComponents/Datalock/SFA.DAS.CollectionEarnings.DataLock/DataLockTask.cs:15]: is there a reason to hard code the timeout here?

File: src/SharedPipelineComponents/Datalock/SFA.DAS.CollectionEarnings.DataLock/DataLockProcessor.cs [hunk 8]
```diff
@@ -37,6 +37,7 @@ protected DataLockProcessor()
...
         {
             _logger.Info("Started Data Lock Processor.");
+            _logger.Info($"Using timeout of: {DataLockTask.CommandTimeout}");
```
Comment (davehowlett) [inline src/SharedPipelineComponents/Datalock/SFA.DAS.CollectionEarnings.DataLock/DataLockProcessor.cs:38]: is there a reason to hard code the timeout here? (the static CommandTimeout is hardcoded)

File: src/SharedPipelineComponents/Datalock/SFA.DAS.CollectionEarnings.DataLock/Infrastructure/Data/Repositories/ValidationErrorRepository.cs [hunk 9]
```diff
@@ -24,7 +24,7 @@ public void AddValidationError(ValidationErrorEntity validationError)
                 AimSeqNumber = validationError.AimSeqNumber,
```
Comment (davehowlett) [inline src/SharedPipelineComponents/Datalock/SFA.DAS.CollectionEarnings.DataLock/Infrastructure/Data/Repositories/ValidationErrorRepository.cs:24]: the dynamic object doesn't need the named fields (e.g. Ukprn, LearnRefNumber)

**Review by davehowlett — CHANGES_REQUESTED** [review]
can you have a look at the missing GO statements and the file to be removed.

**Review by davehowlett — APPROVED** [review]
Solutions now build and the script changes look ok. Minimum acceptance tests all pass as do all integration and unit tests

**Comment by satya-sanikommu** [comment]
Testing completed, no regression issues. Ready to merge

**Pull request merged**